src/spectrum_assignment.py

```diff
@@ -198,3 +273,36 @@ def get_spectrum(self, mod_format_list: list, slice_bandwidth: str = None):
 
             self.spectrum_props.block_reason = 'congestion'
             continue
+
+
+    def get_spectrum_dynamic_slicing(self, mod_format_list: list, slice_bandwidth: str = None, path_index: int = None):
+        """
+        Controls the class, attempts to find an available spectrum.
+
+        :param mod_format_list: A list of modulation formats to attempt allocation.
+        :param slice_bandwidth: A bandwidth used for light-segment slicing.
+        """
+        self._init_spectrum_info()
+
+        if self.engine_props['fixed_grid']:
+            self.spectrum_props.slots_needed = 1
+            self._get_spectrum()
+            if self.spectrum_props.is_free:
+                mod_format, bw, snr_val = self.snr_obj.handle_snr_dynamic_slicing(path_index)
+                if bw == 0:
+                    self.spectrum_props.is_free = False
+                    self.sdn_props.block_reason = "xt_threshold"
+                else:
+
+                    self.spectrum_props.modulation = mod_format
+                    self.spectrum_props.xt_cost = snr_val
+                    self.spectrum_props.is_free = True
+                    self.sdn_props.block_reason = None
+                return mod_format, bw
+            else:
```

Comment:
We might want comments here to understand why zeros are being returned or raising a warning/error.

Reply:
0 means lightpath can not be accommodated